statsmodels/multivariate/factor.py

```diff
@@ -0,0 +1,330 @@
+from statsmodels.iolib import summary2
+
+
+class Factor(Model):
+    """
+    Factor analysis
+
+    .. [1] Hofacker, C. (2004). Exploratory Factor Analysis, Mathematical Marketing.
+    http://www.openaccesstexts.org/pdf/Quant_Chapter_11_efa.pdf
+
+    Supported rotations:
+        'varimax', 'quartimax', 'biquartimax', 'equamax', 'oblimin',
+        'parsimax', 'parsimony', 'biquartimin', 'promax'
+
+    Parameters
+    ----------
+    endog : array-like
+        Variables in columns, observations in rows
+        Could be `None` if `corr` is not `None`
+    n_factor : int
+        The number of factors to extract
+    corr : array-like
+        Directly specify the correlation matrix instead of estimating from endog
+        If not `None`, `endog` will not be used
+    method : str
+        Specify the method to extract factors
+        'pa' - Principal axis factor analysis
+    smc : True or False
+        Whether or not to apply squared multiple correlations
+    endog_names: str
+        Names of endogeous variables.
+        If specified, it will be used instead of the column names in endog
+
+    """
+    def __init__(self, endog, n_factor, corr=None, method='pa', smc=True,
+                 missing='drop', endog_names=None):
+        # CHeck validity of n_factor
+        if n_factor <= 0:
+            raise ValueError('n_factor must be larger than 0! %d < 0' %
+                             (n_factor))
+        if endog is not None and n_factor > endog.shape[1]:
+            raise ValueError('n_factor must be smaller or equal to the number'
+                             ' of columns of endog! %d > %d' %
+                             (n_factor, endog.shape[1]))
+        self.n_factor = n_factor
+
+        if corr is None and endog is None:
+            raise ValueError('Both endog and corr is None!')
+
+        self.loadings = None
+        self.communality = None
+        self.eigenvals = None
+        self.method = method
+        self.smc = smc
+
+        # Check validity of corr
+        if corr is not None:
+            if corr.shape[0] != corr.shape[1]:
+                raise ValueError('Correlation matrix corr must be a square')
+            if endog is not None and endog.shape[1] != corr.shape[0]:
+                    raise ValueError('The number of columns in endog must be '
+                                     'equal to the number of columns and rows corr')
+        self.corr = corr
+        self.endog_names = endog_names
+
+        # Do not preprocess endog if None
+        if endog is not None:
+            super(Factor, self).__init__(endog, exog=None, missing=missing)
+        else:
+            self.endog = None
+
+    @property
+    def endog_names(self):
+        """Names of endogenous variables"""
+        if self._endog_names is not None:
+            return self._endog_names
+        else:
+            if self.endog is not None:
+                return self.data.ynames
+            else:
+                return None
```

Comment:
as default if not available you could make up some names
(that's what we usually do by default in the models)